schema.ts

```diff
@@ -1135,6 +1134,103 @@ export interface IndexDefinition {
   popularityRankProperty?: PropertyIdentifier<string>;
 }
 
+/**
+ * Determines how permissions are handled for this object.
+ */
+export enum PermissionsBehavior {
+  /**
+   * The object will inherit permissions from its parent.
+   */
+  Inherit = 'Inherit',
+}
+
+export interface BaseParentDefinition {
+  /**
+   * The identity of the parent object.
+   */
+  parentIdentity: IdentityDefinition;
+
+  /**
+   * Determines how permissions are handled for this object.
+   */
+  permissions?: PermissionsBehavior;
+}
+
+/**
+ * Defines that instances of this object will not be generated by crawling parents. Instead, parents for this
+ * will be specified in the {@link NotCrawlableParentDefinition.parentIdProperty}.
+ */
+export interface NotCrawlableParentDefinition extends BaseParentDefinition {
+  /**
+   * The name of the property within {@link ObjectSchemaDefinition.properties} that
+   * identifies the parent of this object.
+   *
+   * ValueType.Array is not supported is not supported for parentIdProperty.
+   *
+   * If this property references a property with {@link ValueType.Object} and
+   * {@link ValueHintType.Reference}, then it refers to an object in another logical table.
+   *
+   * Otherwise, it refers to an object in the same table.
+   */
+  parentIdProperty: PropertyIdentifier<string>;
+}
+
+export interface ParentChildParameterMapping {
```

Comment:
you're saying that the _output_ schema of a table will define how the _inputs_ of itself should be populated? that seems kinda loopy - outputs should depend on inputs generally. cc @jonathan-codaio

Reply:
Sorry can you say a bit more? Is the issue that in a dynamic table case, the output of getSchema will influence how the parameters are interpreted, when normally the parameter values themselves can be used as inputs to `getSchema`? That does seem confusing. But in another perspective, should dynamic tables be allowed to participate in crawling? If so, is there any other way to specify that other than waiting until getSchema in order to express the parent identity using a dynamicUrl?

Reply:
It's true that we haven't yet had a need to support dynamic tables in our crawls, but I'm nervous to make that harder to support. The current pattern of having the crawl parent defined on the parameters can work just fine for dynamic tables, once we bother to support it on the `coda` side.

Maybe we should think of this like identities - internally we modeled identities as part of the output schema, but later needed to allow them to be set at the table level via `identityName`. We even made `identityName` required. Maybe this whole `parent table` concept should not be under _either_ parameters or output schema, but rather be its own top level concept. cc @ekoleda-codaio

Reply:
I understand the concern re inversion here, but I think it only really matters in the context of dynamic tables (otherwise schema is staticish). And I don't really understand how crawling would work for dynamic tables anyway. (If there was a way of doing dynamic table crawling, I think it'd have to be a new thing we build).

Reply:
I'm pretty concerned even outside of dynamic sync tables. I made a catalyst for tomorrow to give us time to talk things through - https://coda.io/d/_dCum5Z63Pkj#Topics_tucMmc8a/r1935&view=full